### PR TITLE
More properly handle errors from KCL

### PR DESCRIPTION
**Pull request by xavi-**
Currently in production, consumer are not gracefully handling a lost lease.  These changes should help alleviate those issues.

- Synchronized shutdown checkpointer requests with other line reading/handling action.
- Less aggressively handled errors when parsing kcl messages.

### CODE REVIEW

File: kcl/kcl.go [hunk 1]
```diff
@@ -275,7 +277,6 @@ func (kclp *KCLProcess) Run() {
 		err = kclp.handleLine(line)
 		if err != nil {
 			kclp.ioHandler.writeError(fmt.Sprintf("ERR Handle line: %+#v", err))
-			return
```
Comment (xavi-) [inline kcl/kcl.go:278]: Based on the [python implement](https://github.com/awslabs/amazon-kinesis-client-python/blob/master/samples/sample_kclpy_app.py#L66-L72) it seems like most errors shouldn't cause the process to exit.

**Review by rgarcia — APPROVED** [review]
how did you discover this change needed to happen?

**Comment by xavi-** [comment]
I was looking at the [logs of a crashed container](https://production--haproxy-logs.int.clever.com/_plugin/kibana/?#/discover?_g=(refreshInterval:(display:Off,pause:!f,section:0,value:0),time:(from:'2017-08-16T23:00:31.672Z',mode:absolute,to:'2017-08-17T23:23:43.390Z'))&_a=(columns:!(rawlog),filters:!((meta:(disabled:!f,index:%5Blogs-%5DYYYY-MM-DD,key:container_task,negate:!f,value:'771cca11-a326-4a0f-b615-b6ac38032ecc'),query:(match:(container_task:(query:'771cca11-a326-4a0f-b615-b6ac38032ecc',type:phrase))))),index:%5Blogs-%5DYYYY-MM-DD,interval:auto,query:(query_string:(analyze_wildcard:!t,lowercase_expanded_terms:!f,query:'container_app:kinesis-analytics-consumer')),sort:!(timestamp,desc))) and I noticed that a `"Encountered shutdown exception, skipping checkpoint"` error occurred.  I googled for that error and found [the python implementation](https://github.com/awslabs/amazon-kinesis-client-python/blob/master/samples/sample_kclpy_app.py#L66-L72).  It didn't look like the python code exited the process when it got this error.  I looked at our code and realized that our code did.

**Comment by xavi-** [comment]
As for the `Shutdown` change, I realized it was inherently racy, so I coordinated it's with the rest of the read/handle line code.

**Pull request merged**